source/particle/property/integrated_strain.cc

```diff
@@ -0,0 +1,116 @@
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/property/integrated_strain.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      template <int dim>
+      void
+      IntegratedStrain<dim>::initialize_one_particle_property(const Point<dim> &,
+                                                              const Vector<double> &,
+                                                              const std::vector<Tensor<1,dim> > &,
+                                                              std::vector<double> &data) const
+      {
+        for (unsigned int i = 0; i < Tensor<2,dim>::n_independent_components ; ++i)
+          data.push_back(0.0);
+      }
+
+      template <int dim>
+      void
+      IntegratedStrain<dim>::update_one_particle_property(const unsigned int data_position,
+                                                          const Point<dim> &,
+                                                          const Vector<double> &,
+                                                          const std::vector<Tensor<1,dim> > &gradients,
+                                                          std::vector<double> &data) const
+      {
+        Tensor<2,dim> old_strain;
+        for (unsigned int i = 0; i < Tensor<2,dim>::n_independent_components ; ++i)
+          old_strain[Tensor<2,dim>::unrolled_to_component_indices(i)] = data[data_position + i];
+
+        Tensor<2,dim> grad_u;
+        for (unsigned int d=0; d<dim; ++d)
+          grad_u[d] = gradients[d];
+
+        // Note: We use the old_timestep, because this function is called as
+        // postprocessor right now, and get_timestep() is already updated for
+        // the next timestep.
+        const double dt = this->get_old_timestep();
+
+        // Here we update the integrated strain by rotating the already existing
+        // strain with the rotational (asymmetric) part of the velocity gradient
+        // tensor, and afterwards add the additionally experienced deformation.
+        const Tensor<2,dim> deformation_rate (symmetrize(grad_u));
+
+        // rotation tensor =
+        //     asymmetric part of the displacement in this time step
+        //                (= time step * (velocity gradient tensor - symmetric_part))
+        //   + unit tensor
+        const Tensor<2,dim> rotation (dt * (grad_u - deformation_rate) + unit_symmetric_tensor<dim>());
+
+        // the new strain is the rotated old strain plus the
+        // strain of the current time step
+        const Tensor<2,dim> new_strain = rotation * old_strain * transpose(rotation) + dt * deformation_rate;
```

Comment:
This would be the place. If `old_strain` was just stored as a symmetric tensor, then you may have to do
```
  // convert old strain into a tensor so that it can be multiplied from the left and right by a non-symmetric tensor
  Tensor<2,dim> tmp = rotation * Tensor<2,dim>(old_strain) * transpose(rotation);
  // we know that the result must be a symmetric tensor, so convert that back and add this time step's deformation
  const Tensor<2,dim> new_strain = symmetrize(tmp) + deformation_rate * dt;
```